### PR TITLE
Bmt2 78 product search/filter

### PR DESCRIPTION
**Pull request by MesropMkhitaryan**

### CODE REVIEW

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductSearchServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    public List<Product> searchForProduct(ProductFilterRequest productFilterRequest) {
+        log.info("request filtered product");
+
+        QProduct qProduct = QProduct.product;
+
+        var jpaQuery = new JPAQuery(entityManager);
+
+        var from = jpaQuery.from(qProduct);
+
+        if (productFilterRequest.getTitle() != null && !productFilterRequest.getTitle().equals("")){
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductSearchServiceImpl.java]: ba vor equals exav " "_i ?
Stringi het ashxatelu hamar ogtagorce apach.lang3.StringUtits.isNotBlank

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductSearchServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            from.where(qProduct.title.contains(productFilterRequest.getTitle()));
+        }
+        if (productFilterRequest.getDescription() != null && !productFilterRequest.getDescription().equals("")){
```
Comment (Mushegh128) [inline black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductSearchServiceImpl.java]: StringUtils.isNotBlank()

File: black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java [hunk 3]
```diff
@@ -23,12 +27,29 @@ public class ProductEndpoint {
...
     private final ProductMapper productMapper;
+    private final ProductSearchService productSearchService;
 
     @GetMapping
     public List<ProductResponse> getAllProducts(@PageableDefault(sort = {"createAt"}, direction = Sort.Direction.DESC) Pageable pageable) {
         return productService.findAll(pageable);
     }
 
+    @GetMapping("/search")
+    public List<ProductResponse> searchProduct(@RequestBody ProductFilterRequest productFilterRequest) {
```
Comment (Mushegh128) [inline black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java]: Get zapros@ Body chuni, krnam entadrem or es mas@ run erac u pordzac che. poxe brat

File: black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java [hunk 4]
```diff
@@ -23,12 +27,29 @@ public class ProductEndpoint {
...
+        List<Product> searchedProducts = productSearchService.searchForProduct(productFilterRequest);
+        return searchedProducts.stream()
+                .map(productMapper::toResponse)
+                .collect(Collectors.toCollection(LinkedList::new));
+    }
+
+    @GetMapping("/filtered")
+    public List<ProductResponse> getAllFilteredProducts(@RequestBody ProductFilterRequest productFilterRequest) {
```
Comment (Mushegh128) [inline black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java]: stexel body petq che

**Review by Mushegh128 — CHANGES_REQUESTED** [review]

File: black-mulberry-core/pom.xml [hunk 5]
```diff
@@ -72,25 +76,4 @@
...
-            <version>1.1.3</version>
-            <executions>
-                <execution>
-                    <phase>generate-sources</phase>
-                    <goals>
-                        <goal>process</goal>
-                    </goals>
-                    <configuration>
-                        <outputDirectory>target/generated-sources/java</outputDirectory>
-                        <processor>com.querydsl.apt.jpa.JPAAnnotationProcessor</processor>
```
Comment (Mushegh128) [inline black-mulberry-core/pom.xml]: @hovnaz  sik naye chishte or jnjvele ?

File: black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductSearchServiceImpl.java (hunk 1 above)
Comment (hovnaz) [reply black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductSearchServiceImpl.java]: dra hamar prosty chen dne artagre. hasknalovvvvv

File: black-mulberry-data-transfer/src/main/java/com/black/mulberry/data/transfer/request/ProductFilterRequest.java [hunk 6]
```diff
@@ -1,12 +1,17 @@
...
 import lombok.Data;
 
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+
 @Data
 public class ProductFilterRequest {
-
+    @NotEmpty
```
Comment (Mushegh128) [inline black-mulberry-data-transfer/src/main/java/com/black/mulberry/data/transfer/request/ProductFilterRequest.java]: Stringi hamar ogtagoce @NotBlank.
bayc spes moment -> ProductServiceImpl.searchForProduct_i miji StringUtis.(productFilterRequest.getTitle()) stugumd misht TRUE kuda vortev ed request modeld controlleri mej validaciya kancni.

myam veranaye ed pah@

File: black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java [hunk 7]
```diff
@@ -35,15 +35,15 @@ public List<ProductResponse> getAllProducts(@PageableDefault(sort = {"createAt"}
     }
 
     @GetMapping("/search")
-    public List<ProductResponse> searchProduct(@RequestBody ProductFilterRequest productFilterRequest) {
+    public List<ProductResponse> searchProduct(ProductFilterRequest productFilterRequest) {
```
Comment (Mushegh128) [inline black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java]: anotacian jnjelov Get metod@ body chi unena , kam poxe Post dardzra (vorn or lav gaxapar che , bayc lucum e) , kam inch data or petq e @RequestParamov vercra

u ete guzes Request modeld validacia ancni javax>validationi koxmic kam spring validaciayi koxmic bdi avelcnes @Valid anotaciyan metodi mej 

EX: @PostMapping("/lyalyalya")
	public ResponseEntity<ResponseModel> sayLya(@Valid @NotNull @RequestBody Requestrequest) {
}

u kantralyori anotacvi org.springframework.validation.annotation.Validated anotacian or request metodit miji validacnox anotacianer@ ashxati. 

anpayman sonq eneluc heto myam postmanov zapros era debag rejimov u hamozvi or 400 kstanas + breackpontd chi erta araj(nu chbdi brne)

File: black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java [hunk 8]
```diff
@@ -35,15 +35,15 @@ public List<ProductResponse> getAllProducts(@PageableDefault(sort = {"createAt"}
...
         List<Product> searchedProducts = productSearchService.searchForProduct(productFilterRequest);
         return searchedProducts.stream()
                 .map(productMapper::toResponse)
                 .collect(Collectors.toCollection(LinkedList::new));
     }
 
-    @GetMapping("/filtered")
-    public List<ProductResponse> getAllFilteredProducts(@RequestBody ProductFilterRequest productFilterRequest) {
+    @GetMapping("/filter")
+    public List<ProductResponse> getAllFilteredProducts(ProductFilterRequest productFilterRequest) {
```
Comment (Mushegh128) [inline black-mulberry-rest/src/main/java/com/black/mulberry/rest/endpoint/ProductEndpoint.java]: -

**Review by Mushegh128 — CHANGES_REQUESTED** [review]

File: black-mulberry-core/pom.xml (hunk 5 above)
Comment (MesropMkhitaryan) [reply black-mulberry-core/pom.xml]: ha sig pti jnjver

**Pull request merged**